models/nc-news-models.js

```diff
@@ -62,6 +65,29 @@ function addCommentByArticleById(article_id, commentData) {
     )
     .then(({ rows }) => {
       return rows[0];
+    })
+    .catch((err) => {
+      //console.log(err);
+      /// I had to Rethrow the error to be caught in the calling function
+      throw err;
+    });
+}
+
+function updateArticleByArticleId(article_id, artData, currVotes) {
+  const { inc_votes } = artData;
+  const updatedVotes = currVotes + inc_votes;
+  return db
+    .query(
+      `UPDATE articles SET votes = $1 WHERE article_id = $2 RETURNING *;`,
+      [updatedVotes, article_id]
+    )
+    .then(({ rows }) => {
+      return rows[0];
+    })
+    .catch((err) => {
+      console.log(err);
+      /// I had to Rethrow the error to be caught in the calling function
```

Comment:
This should be handled in your controller, you would catch the error using .catch(next) and pass it to the error-handling middleware.